File: 02. Algorithms/04. Recursion/Problems/03.Towers of Hanoi/TowerOfHanoi.java
public class TowerOfHanoi {
    public static void main(String[] args) {
        towerOfHanoi(3, 'A','B', 'C');
    }

    static void towerOfHanoi(int n, char from, char aux, char to) {
        if (n>0) {
            towerOfHanoi(n-1, from, to, aux);
            System.out.println("Move " + n + " from " + from + " to " + to );
            towerOfHanoi(n-1, aux, from, to);
        }
    }
}

File: 02. Algorithms/04. Recursion/Problems/03.Towers of Hanoi/TowerOfHanoi.py
# Creating a recursive function  
def Towerofhanoi(disks, source, auxiliary, target):  
    if(disks == 1):  
        print('Move disk 1 from rod {} to rod {}.'.format(source, target))  
        return  
    # function call itself  
    Towerofhanoi(disks - 1, source, target, auxiliary)  
    print('Move disk {} from rod {} to rod {}.'.format(disks, source, target))  
    Towerofhanoi(disks - 1, auxiliary, source, target)  
disks = int(input('Enter the number of disks: '))  
# We are referring source as A, auxiliary as B, and target as C  
Towerofhanoi(disks, 'A', 'B', 'C')  # Calling the function  
